src/pybind/rados/rados.pyx

```diff
@@ -2370,6 +2371,52 @@ cdef class Ioctx(object):
             raise make_ex(ret, "error writing object %s" % object_name)
         return completion
 
+    @requires(('object_name', str_type), ('to_write', bytes), ('write_len', int),
+              ('offset', int), ('oncomplete', opt(Callable)),
+              ('onsafe', opt(Callable)))
+    def aio_writesame(self, object_name, to_write, write_len, offset=0,
+                      oncomplete=None, onsafe=None):
+        """    
+        Asynchronously write the same buffer multiple times
+
+        :param object_name: name of the object
+        :type object_name: str
+        :param to_write: data to write
+        :type to_write: bytes
+        :param write_len: total number of bytes to write
+        :type write_len: int
+        :param offset: byte offset in the object to begin writing at
+        :type offset: int
+        :param oncomplete: what to do when the writesame is safe and 
+            complete in memory on all replicas
+        :param onsafe: what to do when the writesame is safe and complete
```

Comment:
we've deprecated onsafe callback. so you might want to remove it from this method and pass `None` to `self.__get_completion()`?

Reply:
yeah yeah yeah

Reply:
I just pass none to self.__get_completion() . I don't know what needs to be updated

Reply:
@ideepika  Updated, please review.

Reply:
@ideepika  please check, very very very very thankful.